### PR TITLE
Update for 4.10

### PR DESCRIPTION
**Pull request by eskatos**
- update samples markup to use `include::sample` macro and account for recent style changes
- add coverage of the configuration avoidance API & DSL sugar
- move samples testing to Examplar

### CODE REVIEW

**Comment by eskatos** [comment]
@eriwen I'm going to do another pass tomorrow but I'd appreciate your early feedback if you can

File: src/test/groovy/org/gradle/guides/SamplesTest.groovy [hunk 1]
```diff
@@ -12,75 +12,20 @@ class SamplesTest extends AbstractSamplesFunctionalTest {
         initialize()
     }
 
-    @Unroll
```
Comment (eriwen) [inline src/test/groovy/org/gradle/guides/SamplesTest.groovy]: Aha, nice right? ;)

Comment (eskatos) [reply src/test/groovy/org/gradle/guides/SamplesTest.groovy]: Indeed 👍

File: contents/index.adoc [hunk 2]
```diff
@@ -311,84 +314,97 @@ Now that we know the types of the two tasks, we can import the relevant types 
...
-[source.multi-language-sample, groovy]
-----
-include::{samplescodedir}/configurations-and-dependencies/declarative.gradle[]
-----
-
-[source.multi-language-sample, kotlin]
-----
-include::{samplescodedir}/configurations-and-dependencies/declarative.gradle.kts[]
-----
+====
```
[the remark this answers is not in the document]
Comment (eskatos) [reply contents/index.adoc]: Yep, intentional

File: contents/index.adoc [hunk 3]
```diff
@@ -210,64 +185,92 @@ The same approach can be used to resolve plugins from composite builds, which li
...
+Gradle 4.9 introduced a new API for creating and configuring tasks in build scripts and plugins. The intent is for this new API to eventually replace the existing API.
+
+[quote]
+____
+One of the major differences between the existing and new Gradle Tasks API is whether or not Gradle spends the time to create `Task` instances and run configuration code. The new API allows Gradle to delay or completely avoid configuring tasks that will never be executed in a build. For example, when compiling code, Gradle does not need to configure tasks that run tests.
+____
+
+See the link:https://blog.gradle.org/preview-avoiding-task-configuration-time[_Evolving the Gradle API to reduce configuration time_] blog post and the link:{user-manual}task_configuration_avoidance.html[Task Configuration Avoidance] chapter in the user manual for more information.
+
+The Gradle Kotlin DSL embraces the new API by making the DSL sugar use it by default. Rest assured, the whole Gradle API remains available.
```
[the remark this answers is not in the document]
Comment (eskatos) [reply contents/index.adoc]: Good point, thanks, I rephrased that sentence.

**Pull request merged**